### PR TITLE
CONTOUR change smoothing mode and smoothing factor test

### PR DESCRIPTION
**Pull request by acdo2002**
The original name is PER_CONTOUR_DATA_2D.test.ts
Now I rename it as CONTOUR_CHANGE_SMOOTH_MODE_FACTOR.test.ts
And I add the doc in [here](https://docs.google.com/document/d/1RvTIH2664S7jBYEvaTm_D5Pd07S5cTmcL7VJlCU0Uxw/edit#)

### CODE REVIEW

**Comment by kswang1029** [comment]
by changing mode and rounding factor, you need to verify the contour vertices (thus not a good idea using random number).

**Comment by kswang1029** [comment]
> by changing mode and rounding factor, you need to verify the contour vertices (thus not a good idea using random number).

CONTOUR_IMAGE_DATA is to change different smoothing mode. So it has been tested. New test should address smoothing_factor and decimation_factor, and verify contour vertices.

**Comment by acdo2002** [comment]
I have changed the tested based on the KS suggestion.
The test only change smoothing_factor and decimation_factor and verify the vertices via snapshot. 
@zarda and @kswang1029 please review it again. Thanks!!!

File: src/test/CONTOUR_CHANGE_SMOOTH_AND_DEMICATION_FACTOR.test.ts [hunk 1]
```diff
@@ -102,38 +121,27 @@ describe("PER_CONTOUR_DATA_2D", () => {
...
-                    (type, data, ack) => ack.ContourImageData.filter(data => data.progress == 1).length == assertItem.setContour[0].levels.length
-                );
-            }, contourTimeout);
-
-            let SF_number = new Array(5).fill(1).map((_, i) => i + 1);
-            let newnumber = SF_number.sort(() => Math.random() - 0.5);
-            console.log('New Contour smoothing factor (random between 1 and 5):', newnumber);
+            let ContourImageData: CARTA.ContourImageData;
+            assertItem.setContour.map((input,index)=>{
+                test(`(Step ${index+2}: Set Smoothing factor of ${input.smoothingFactor} & Decimation factor of ${input.decimationFactor}): Check Vertices coordinates are consistent to Snapshot`,async()=>{
```
Comment (zarda) [inline src/test/CONTOUR_CHANGE_SMOOTH_AND_DEMICATION_FACTOR.test.ts:133]: May need a timeout to handle a slow network environment

**Review by zarda — COMMENTED** [review]
Please refactor the code line 125 by `Ack = await Connection.streamUntil((type, data) => type == CARTA.RasterTileSync && data.endSync );`
Thanks

**Comment by acdo2002** [comment]
@zarda I have modify the Ack as well as add contour timeout to the test.

**Review by zarda — APPROVED** [review]

**Pull request merged**